### DISCUSSION TITLE
'connect wallet' button text as prop

### DISCUSSION THREAD

**Discussion by nerderlyne** [rainbow-me/rainbowkit#346] [Ideas]
This currently doesn't work 
`<ConnectButton>Connect</ConnectButton>`

**Comment by nickbytes** [comment]
If you want to customize the experience of the Connect Button, you could try using our [Custom Connect Button feature](https://www.rainbowkit.com/docs/custom-connect-button). With a custom Connect Button, you get control over everything in your UI (besides the modals.)

**Comment by peduarte** [comment]
What nick said ☝️ 

A bit more context: The `ConnectButton` technically doesn't just render a button. It renders different things depending on the connected state:

When disconnected:
- A `button` with a "Connect wallet" string

When connected:
- A network switcher button
- An account button containing
  - Account balance
  - Address
  - ENS name
  - Avatar

I am implying from your message that what you'd like is a simple way to only update the disconnected button's text, while maintaining all of the other styles/functionality. 

I'm open to chatting about this, and hearing other people's thoughts too. 

We'd need an API that makes it clear about what you're editing though. Would you like to provide some examples?

Thanks!
